acceptance/tests/catalog_with_binary_data.rb

```diff
@@ -0,0 +1,72 @@
+require 'puppet/acceptance/common_utils'
+require 'puppet/acceptance/environment_utils'
+extend Puppet::Acceptance::EnvironmentUtils
+
+test_name "C100300: Catalog containing binary data is applied correctly" do
+  skip_test 'requires a master for serving module content' if master.nil?
+  tag 'risk:medium'
+
+  test_num = 'c100300'
+  tmp_environment = mk_tmp_environment_with_teardown(master, File.basename(__FILE__, '.*'))
+  agent_tmp_dirs = {}
+  agents.each do |agent|
+    agent_tmp_dirs[agent.hostname] = agent.tmpdir(tmp_environment)
+  end
+
+  teardown do
+    step 'remove all test files on agents' do
+      agents.each { |agent| on(agent, "rm -r #{agent_tmp_dirs[agent.hostname]}", :accept_all_exit_codes => true) }
+    end
+    # note - master teardown is registered by #mk_tmp_environment_with_teardown
+  end
+
+  step "Create module with binary data file on master" do
+    on(master, "mkdir -p #{environmentpath}/#{tmp_environment}/modules/#{test_num}/{manifests,files}")
+    master_module_manifest = "#{environmentpath}/#{tmp_environment}/modules/#{test_num}/manifests/init.pp"
+    master_module_binary_file = "#{environmentpath}/#{tmp_environment}/modules/#{test_num}/files/binary_data"
+
+    bin_content = '\xC0\xFF'
```

Comment:
@johnduarte I think we want this to be wrapped in double quotes, not single quotes. If wrapped in single quotes, ruby doesn't evaluate the escape sequence to evaluate as hex value.:
```ruby
[8] pry(main)> bin = "\xC0\xFF"
=> "\xC0\xFF"
[9] pry(main)> bin.valid_encoding?
=> false
[10] pry(main)> bin = '\xC0\xFF'
=> "\\xC0\\xFF"
[11] pry(main)> bin.valid_encoding?
=> true
```